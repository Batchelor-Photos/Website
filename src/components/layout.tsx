/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import React, { FC, ReactNode } from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Typography from '@material-ui/core/Typography'
import { ThemeProvider } from '@material-ui/core/styles'

import { login, isAuthenticated } from '../utils/auth'
import gatsbyTheme from '../mui-themes/gatsbyTheme'
import Header from './header'
import './layout.css'

interface LayoutProps {
  children: ReactNode
  requireAuth?: boolean
}

interface PureLayoutProps {
  children: ReactNode
  data: {
    site: {
      siteMetadata: {
        title: string
      }
    }
  }
}

export const Layout: FC<LayoutProps> = ({ requireAuth = true, ...props }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  if (requireAuth && !isAuthenticated()) {
    login({ redirectUrl: window.location.pathname })
    return null
  }

  return <PureLayout data={data} {...props} />
}

export const PureLayout: FC<PureLayoutProps> = ({ children, data }) => {
  return (
    <ThemeProvider theme={gatsbyTheme}>
      <Header siteTitle={data.site.siteMetadata.title} />
      <div
        style={{
          margin: `0 auto`,
          maxWidth: 960,
          padding: `0px 1.0875rem 1.45rem`,
          paddingTop: 0,
        }}
      >
        <main>{children}</main>
        <footer>
          <Typography variant="body1">
            Â© {new Date().getFullYear()}, Built with <a href="https://www.gatsbyjs.org">Gatsby</a>
          </Typography>
        </footer>
      </div>
    </ThemeProvider>
  )
}

export default Layout
