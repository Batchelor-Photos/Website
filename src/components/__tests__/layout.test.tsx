import React from 'react'
import renderer from 'react-test-renderer'
import { PureLayout as Layout } from '../layout'

describe('Layout', () => {
  it('renders correctly', () => {
    const data = {
      site: {
        siteMetadata: {
          title: 'Test Title',
        },
      },
    }

    const tree = renderer
      .create(
        <Layout data={data}>
          <p>Testing 123</p>
        </Layout>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
