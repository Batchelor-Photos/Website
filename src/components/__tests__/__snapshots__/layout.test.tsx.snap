// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <header
    style={
      Object {
        "backgroundColor": "#663399",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      className="MuiBox-root MuiBox-root-2"
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem",
        }
      }
    >
      <h1
        className="MuiTypography-root MuiTypography-h4"
        style={
          Object {
            "fontWeight": 700,
            "margin": "0 auto",
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "#fff",
              "textDecoration": "none",
            }
          }
        >
          Test Title
        </a>
      </h1>
    </div>
  </header>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0px 1.0875rem 1.45rem",
        "paddingTop": 0,
      }
    }
  >
    <main>
      <p>
        Testing 123
      </p>
    </main>
    <footer>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Â© 
        2019
        , Built with 
        <a
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
      </p>
    </footer>
  </div>,
]
`;
